### PR TITLE
Bump tektoncd/pipeline to the latest release 0.10.0… 🐸

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

… alongside with the k8s.io/* dependencies as tektoncd/pipeline tracks
1.13.4 now.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/cc @a-roberts @CarolynMabbott 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by a-roberts** [comment]
@vdemeester thanks for this 😸 do you know if that field's available in 0.8 too? We like the dashboard supporting both 0.8 and 0.9 atm

I'm sure TestWebsocketResources isn't related to this PR, may be another flaky one 😸

**Comment by vdemeester** [comment]
> @vdemeester thanks for this :smile_cat: do you know if that field (for retrieving info) is available in 0.8 too? We like the dashboard supporting both 0.8 and 0.9 atm

@a-roberts do you mean `TaskRunStatusFields` ? It has `json:",inline"` so nothing change on the marshall object (yaml or json), so dashboard will support 0.8 and 0.9 for sure :wink:

**Comment by a-roberts** [comment]
> > @vdemeester thanks for this 😸 do you know if that field (for retrieving info) is available in 0.8 too? We like the dashboard supporting both 0.8 and 0.9 atm
> 
> @a-roberts do you mean `TaskRunStatusFields` ? It has `json:",inline"` so nothing change on the marshall object (yaml or json), so dashboard will support 0.8 and 0.9 for sure 😉

Perfect, yep that's what I meant, let's merge once the build is green 😄

**Comment by a-roberts** [comment]
Discussed this one with @mnuttall and @CarolynMabbott, so what we're going to do is

- Pin to Tekton Pipelines 0.8
- Do a release of the dashboard and extension (0.4.0)
(this is important for a Kabanero release)
- Then move up to 0.9.2 (and land this PR)
- Then move up to master (we noticed changes for PipelineResources we will need to contain and so we think that's going to be a bigger PR)

Therefore for now, /hold

File: Gopkg.toml [hunk 1]
```diff
@@ -7,18 +7,18 @@ required = [
...
 
 [[override]]
   name = "k8s.io/apimachinery"
-  version = "kubernetes-1.12.6"
+  version = "kubernetes-1.13.4"
 
 [[override]]
   name = "k8s.io/api"
-  version = "kubernetes-1.12.6"
+  version = "kubernetes-1.13.4"
```
Comment (a-roberts) [inline Gopkg.toml]: Having a look again as we think we've done everything for Tekton 0.8, Is this gonna work if we want to still support OCP 4.2 which uses Kube 1.12?

We want to do a 0.5.0 release for Pipelines 0.10 and Triggers 0.2 but for OpenShift 4.2

Comment (vdemeester) [reply Gopkg.toml]: I will work :stuck_out_tongue: 0.9.0 is already depending on 1.13. Depending on this doesn't mean we can't target lower kubernetes version :angel:

Comment (a-roberts) [reply Gopkg.toml]: Excellent, let's get it in then

**Comment by vdemeester** [comment]
Ah I need to fix the conflicts :P

**Comment by vdemeester** [comment]
@CarolynMabbott @a-roberts it brings back the `gopkg.lock` file as it is required to ensure reproductibility in builds (otherwise, whatever is not in the `toml` can differ from time A to time B).

Usually, you change something in the `toml`, and do a `dep ensure -v`, and the `.lock` file gets updated :angel:.

**Comment by vdemeester** [comment]
Ah damn, I forgot to update something in the code.. :sweat: :sweat_smile:

File: test/e2e-common.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ function print_diagnostic_info() {
 
 function install_pipeline_crd() {
   echo ">> Deploying Tekton Pipelines"
-  kubectl apply --filename https://github.com/tektoncd/pipeline/releases/download/v0.8.0/release.yaml || fail_test "Tekton pipeline installation failed"
+  kubectl apply --filename https://github.com/tektoncd/pipeline/releases/download/v0.10.0/release.yaml || fail_test "Tekton pipeline installation failed"
```
Comment (vdemeester) [inline test/e2e-common.sh]: :smiling_imp:

**Comment by a-roberts** [comment]
Yeah, will make a new issue to upgrade integration tests to work on 0.10, this is a good start, I'm aware the Gopkg.lock file is back and have nothing against it really

/lgtm

**Pull request merged**